### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by rben01** [coteditor/CotEditor#1489]
### Description

Background:
I was using VSCode to edit a json file that I also had open in CotEditor. The file was small, around 1KB. Randomly (well, on whatever schedule macOS uses), my computer automatically switched from light mode to dark mode, and immediately everything ground to a halt. Finally, I managed to get to the Memory tab of Activity Monitor and saw that CotEditor was using an ungodly amount of memory: 78GB. The amount of memory was growing seemingly without bound — my computer was using 28GB of swap and that number was increasing at a rate of about 1GB per minute.

Unfortunately I couldn't get any info on the CotEditor process; I had to just kill it outright, at which point everything went back to normal.

<img width="789" alt="Screen Shot 2023-05-24 at 9 24 45 PM" src="https://github.com/coteditor/CotEditor/assets/766670/3d15428d-219e-4089-8698-c9ba8f661b8f">

### To Reproduce

Not really sure what the specific steps were; see above.

### CotEditor version

Version 4.5.5 (569)

### macOS version

12.6.6 (21G646)

**Comment by alfredrajo** [comment]
same here. Happens 1 time a day, not reproducible. Moving back to bbedit

**Comment by guneyozsan** [comment]
Btw, it now happens every time I run it. A couple of minutes is sufficient to fill +50GB of memory.
<img width="420" alt="CleanShot 2023-09-23 at 15 52 30@2x" src="https://github.com/coteditor/CotEditor/assets/19275187/5e768aa0-7121-47e6-a44d-1364edbe4efe">

**Comment by 1024jp** [comment]
I have added multiple measures to the last versions including the latest CotEditor 4.7.0, released yesterday.
However, honestly, I'm unsure whether this leak could be solved.
So, let me know if you face this issue again in the new versions; otherwise, I'll close this ticket after a few months.

**Comment by guneyozsan** [comment]
So far it's going fine (currently version 4.7.3). I'll get back to here if I face it again.

**Comment by 1024jp** [comment]
That's nice to hear!
Since I close this thread, create a new one if you face a similar case again.

**Comment by rben01** [comment]
Appears not to be fixed. This screenshot was taken with two files open, each under 100KB.

<img width="595" alt="04BB4ADC-449B-4024-91EC-746DBAD1D23B" src="https://github.com/coteditor/CotEditor/assets/766670/ed1be99d-0424-4ca6-b518-e19e4504c985">

Here is the crash report info after killing it in Activity Monitor:

```
Date/Time:        2024-02-05 12:59:35.869 -0500
End time:         2024-02-05 12:59:36.969 -0500
OS Version:       macOS 14.3 (Build 23D56)
Architecture:     arm64e
Report Version:   44
Incident Identifier: AB5921C5-9803-4AFE-BD3D-A38887678275
Share With Devs:  Yes

Data Source:      Stackshots
Shared Cache:     B81268A5-00ED-3883-B922-13118E8800FE slid base address 0x181f70000, slide 0x1f70000 (System Primary)
Shared Cache:     3F752D9C-E698-3234-A7DF-3FFFA63125F9 slid base address 0x194dfc000, slide 0x14dfc000 (DriverKit)
Shared Cache:     A8965321-FE05-3C38-AE76-BC877557295B slid base address 0x7ff807064000, slide 0x7064000 (Rosetta)

Command:          CotEditor
Path:             /Applications/CotEditor.app/Contents/MacOS/CotEditor
Identifier:       com.coteditor.CotEditor
Version:          4.7.3 (627)
Team ID:          HT3Z3A72WZ
App Item ID:      1024640650
App External ID:  863050276
Is First Party:   No
Architecture:     arm64
PID:              1166
Time Since Fork:  430868s

Event:            hang
Duration:         1.10s
Steps:            11 (100ms sampling interval)

Hardware model:   Mac14,15
Active cpus:      8
HW page size:     16384
VM page size:     16384

Time Since Boot:  431047s
Time Awake Since Boot: 90740s
Time Since Wake:  335s

Total CPU Time:   1.882s (3.2G cycles, 3.9G instructions, 0.83c/i)
Advisory levels:  Battery -> 3, User -> 2, ThermalPressure -> 0, Combined -> 2
Free disk space:  4112.72 MB/228.27 GB, low space threshold 3072 MB
Vnodes Available: 65.51% (164630/251295)

Preferred User Language: en-US
Country Code:     US
OS Cryptex File Extents: 2036

--------------------------------------------------
Timeline format: stacks are sorted chronologically
Use -i and -heavy to re-report with count sorting
--------------------------------------------------


Heaviest stack for the main thread of the target process:
  11  <truncated backtrace>
  11  start + 2360 (dyld + 24800) [0x18202d0e0]
  11  ??? (CotEditor + 98720) [0x1008e41a0]
  11  NSApplicationMain + 880 (AppKit + 17984) [0x185c33640]
  11  -[NSApplication run] + 476 (AppKit + 185212) [0x185c5c37c]
  11  -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716 (AppKit + 8539884) [0x186453eec]
  11  _DPSNextEvent + 660 (AppKit + 237264) [0x185c68ed0]
  11  _BlockUntilNextEventMatchingListInModeWithFilter + 76 (HIToolbox + 196572) [0x18ca51fdc]
  11  ReceiveNextEventCommon + 648 (HIToolbox + 197252) [0x18ca52284]
  11  RunCurrentEventLoopInMode + 292 (HIToolbox + 197704) [0x18ca52448]
  11  CFRunLoopRunSpecific + 608 (CoreFoundation + 506172) [0x18248993c]
  11  __CFRunLoopRun + 1996 (CoreFoundation + 509792) [0x18248a760]
  11  __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16 (CoreFoundation + 781988) [0x1824ccea4]
  11  _dispatch_main_queue_callback_4CF + 44 (libdispatch.dylib + 72640) [0x18220abc0]
  11  _dispatch_main_queue_drain + 744 (libdispatch.dylib + 73400) [0x18220aeb8]
  11  swift_job_runImpl(swift::Job*, swift::ExecutorRef) + 72 (libswift_Concurrency.dylib + 274356) [0x22360afb4]
  11  swift::runJobInEstablishedExecutorContext(swift::Job*) + 436 (libswift_Concurrency.dylib + 269756) [0x223609dbc]
  11  ??? (CotEditor + 381440) [0x100929200]
  11  ??? (CotEditor + 381896) [0x1009293c8]
  11  ??? (CotEditor + 354060) [0x10092270c]
  11  ??? (CotEditor + 1631460) [0x100a5a4e4]
  11  BidirectionalCollection<>.difference<A>(from:) + 204 (libswiftCore.dylib + 2417160) [0x191ecb208]
  11  _withContiguousStorage #1 <A, B><A1, B1>(for:_:) in _myers<A, B>(from:to:using:) + 672 (libswiftCore.dylib + 2420704) [0x191ecbfe0]
  11  closure #1 in _myers<A, B>(from:to:using:)partial apply + 16 (libswiftCore.dylib + 3383492) [0x191fb70c4]
  11  partial apply for closure #1 in _myers<A, B>(from:to:using:) + 40 (libswiftCore.dylib + 3383540) [0x191fb70f4]
  11  closure #1 in _myers<A, B>(from:to:using:) + 492 (libswiftCore.dylib + 2421304) [0x191ecc238]
  11  closure #1 in closure #1 in _myers<A, B>(from:to:using:) + 128 (libswiftCore.dylib + 2421488) [0x191ecc2f0]
  11  bool swift::RefCounts<swift::RefCountBitsT<(swift::RefCountInlinedness)1>>::doDecrementSlow<(swift::PerformDeinit)1>(swift::RefCountBitsT<(swift::RefCountInlinedness)1>, unsigned int) + 136 (libswiftCore.dylib + 3742348) [0x19200ea8c]
  11  _swift_release_dealloc + 56 (libswiftCore.dylib + 3739056) [0x19200ddb0]
  11  _ContiguousArrayStorage.__deallocating_deinit + 96 (libswiftCore.dylib + 628952) [0x191d168d8]
  8   swift_release + 48 (libswiftCore.dylib + 3727072) [0x19200aee0]
 *8   ??? (kernel.release.t8112 + 30924) [0xfffffe00085678cc]
 *8   ??? (kernel.release.t8112 + 1679204) [0xfffffe00086f9f64]
 *8   ??? (kernel.release.t8112 + 1681884) [0xfffffe00086fa9dc]
 *8   ??? (kernel.release.t8112 + 1047444) [0xfffffe000865fb94]
 *8   ??? (kernel.release.t8112 + 1064764) [0xfffffe0008663f3c]
 *8   ??? (kernel.release.t8112 + 1009828) [0xfffffe00086568a4]
 *8   ??? (kernel.release.t8112 + 998824) [0xfffffe0008653da8]
 *8   ??? (kernel.release.t8112 + 999652) [0xfffffe00086540e4]
 *8   ??? (kernel.release.t8112 + 992140) [0xfffffe000865238c]
 *8   ??? (kernel.release.t8112 + 1029292) [0xfffffe000865b4ac]
 *8   ??? (kernel.release.t8112 + 6180976) [0xfffffe0008b45070]
 *8   ??? (kernel.release.t8112 + 6153540) [0xfffffe0008b3e544]
 *8   apfs_vnop_pagein + 300 (com.apple.filesystems.apfs + 458668) [0xfffffe000b57de1c]
 *8   apfs_pagein + 2004 (com.apple.filesystems.apfs + 460936) [0xfffffe000b57e6f8]
 *8   cluster_pagein + 28 (kernel.release.t8112 + 2238300) [0xfffffe000878275c]
 *8   cluster_pagein_ext + 244 (kernel.release.t8112 + 2238556) [0xfffffe000878285c]
 *8   ??? (kernel.release.t8112 + 2233840) [0xfffffe00087815f0]
 *8   buf_biowait + 104 (kernel.release.t8112 + 2188596) [0xfffffe0008776534]
 *8   ??? (kernel.release.t8112 + 5566384) [0xfffffe0008aaefb0]
 *8   lck_mtx_sleep + 228 (kernel.release.t8112 + 471164) [0xfffffe00085d307c]
 *8   ??? (kernel.release.t8112 + 561416) [0xfffffe00085e9108]
 *8   ??? (kernel.release.t8112 + 567968) [0xfffffe00085eaaa0]


Process:          CotEditor [1166]
UUID:             AE6A14B2-CEB3-3232-8218-26C78637E73A
Path:             /Applications/CotEditor.app/Contents/MacOS/CotEditor
Identifier:       com.coteditor.CotEditor
Version:          4.7.3 (627)
Team ID:          HT3Z3A72WZ
App Item ID:      1024640650
App External ID:  863050276
Is First Party:   No
Shared Cache:     B81268A5-00ED-3883-B922-13118E8800FE slid base address 0x181f70000, slide 0x1f70000 (System Primary)
Architecture:     arm64
Footprint:        13.30 GB -> 8086.47 MB (-5528.53 MB)
Time Since Fork:  430868s
Num samples:      11 (1-11)
CPU Time:         0.108s (330.3M cycles, 607.7M instructions, 0.54c/i)
Note:             1 idle work queue thread omitted

  Thread 0x1fbf    Swift Task UNKNOWN    DispatchQueue "com.apple.main-thread"(1)    11 samples (1-11)    priority 46 (base 46)    cpu time 0.108s (330.3M cycles, 607.7M instructions, 0.54c/i)
  <thread QoS user interactive (requested user interactive), process unclamped, process received importance donation from WindowServer [567], process received importance donation from distnoted [876], process received importance donation from filecoordinationd [1028], IO tier 0>
  11  <truncated backtrace> 1-11
    11  start + 2360 (dyld + 24800) [0x18202d0e0] 1-11
      11  ??? (CotEditor + 98720) [0x1008e41a0] 1-11
        11  NSApplicationMain + 880 (AppKit + 17984) [0x185c33640] 1-11
          11  -[NSApplication run] + 476 (AppKit + 185212) [0x185c5c37c] 1-11
            11  -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716 (AppKit + 8539884) [0x186453eec] 1-11
              11  _DPSNextEvent + 660 (AppKit + 237264) [0x185c68ed0] 1-11
                11  _BlockUntilNextEventMatchingListInModeWithFilter + 76 (HIToolbox + 196572) [0x18ca51fdc] 1-11
                  11  ReceiveNextEventCommon + 648 (HIToolbox + 197252) [0x18ca52284] 1-11
                    11  RunCurrentEventLoopInMode + 292 (HIToolbox + 197704) [0x18ca52448] 1-11
                      11  CFRunLoopRunSpecific + 608 (CoreFoundation + 506172) [0x18248993c] 1-11
                        11  __CFRunLoopRun + 1996 (CoreFoundation + 509792) [0x18248a760] 1-11
                          11  __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16 (CoreFoundation + 781988) [0x1824ccea4] 1-11
                            11  _dispatch_main_queue_callback_4CF + 44 (libdispatch.dylib + 72640) [0x18220abc0] 1-11
                              11  _dispatch_main_queue_drain + 744 (libdispatch.dylib + 73400) [0x18220aeb8] 1-11
                                11  swift_job_runImpl(swift::Job*, swift::ExecutorRef) + 72 (libswift_Concurrency.dylib + 274356) [0x22360afb4] 1-11
                                  11  swift::runJobInEstablishedExecutorContext(swift::Job*) + 436 (libswift_Concurrency.dylib + 269756) [0x223609dbc] 1-11
                                    11  ??? (CotEditor + 381440) [0x100929200] 1-11
                                      11  ??? (CotEditor + 381896) [0x1009293c8] 1-11
                                        11  ??? (CotEditor + 354060) [0x10092270c] 1-11
                                          11  ??? (CotEditor + 1631460) [0x100a5a4e4] 1-11
                                            11  BidirectionalCollection<>.difference<A>(from:) + 204 (libswiftCore.dylib + 2417160) [0x191ecb208] 1-11
                                              11  _withContiguousStorage #1 <A, B><A1, B1>(for:_:) in _myers<A, B>(from:to:using:) + 672 (libswiftCore.dylib + 2420704) [0x191ecbfe0] 1-11
                                                11  closure #1 in _myers<A, B>(from:to:using:)partial apply + 16 (libswiftCore.dylib + 3383492) [0x191fb70c4] 1-11
                                                  11  partial apply for closure #1 in _myers<A, B>(from:to:using:) + 40 (libswiftCore.dylib + 3383540) [0x191fb70f4] 1-11
                                                    11  closure #1 in _myers<A, B>(from:to:using:) + 492 (libswiftCore.dylib + 2421304) [0x191ecc238] 1-11
                                                      11  closure #1 in closure #1 in _myers<A, B>(from:to:using:) + 128 (libswiftCore.dylib + 2421488) [0x191ecc2f0] 1-11
                                                        11  bool swift::RefCounts<swift::RefCountBitsT<(swift::RefCountInlinedness)1>>::doDecrementSlow<(swift::PerformDeinit)1>(swift::RefCountBitsT<(swift::RefCountInlinedness)1>, unsigned int) + 136 (libswiftCore.dylib + 3742348) [0x19200ea8c] 1-11
                                                          11  _swift_release_dealloc + 56 (libswiftCore.dylib + 3739056) [0x19200ddb0] 1-11
                                                            11  _ContiguousArrayStorage.__deallocating_deinit + 96 (libswiftCore.dylib + 628952) [0x191d168d8] 1-11
                                                              3   swift_release + 48 (libswiftCore.dylib + 3727072) [0x19200aee0] 1-3
                                                               *3   ??? (kernel.release.t8112 + 30924) [0xfffffe00085678cc] 1-3
                                                                 *3   ??? (kernel.release.t8112 + 1679204) [0xfffffe00086f9f64] 1-3
                                                                   *3   ??? (kernel.release.t8112 + 1681884) [0xfffffe00086fa9dc] 1-3
                                                                     *3   ??? (kernel.release.t8112 + 1047444) [0xfffffe000865fb94] 1-3
                                                                       *3   ??? (kernel.release.t8112 + 1064764) [0xfffffe0008663f3c] 1-3
                                                                         *3   ??? (kernel.release.t8112 + 1009828) [0xfffffe00086568a4] 1-3
                                                                           *3   ??? (kernel.release.t8112 + 998824) [0xfffffe0008653da8] 1-3
                                                                             *3   ??? (kernel.release.t8112 + 999652) [0xfffffe00086540e4] 1-3
                                                                               *3   ??? (kernel.release.t8112 + 992140) [0xfffffe000865238c] 1-3
                                                                                 *3   ??? (kernel.release.t8112 + 1029292) [0xfffffe000865b4ac] 1-3
                                                                                   *3   ??? (kernel.release.t8112 + 6180976) [0xfffffe0008b45070] 1-3
                                                                                     *3   ??? (kernel.release.t8112 + 6153540) [0xfffffe0008b3e544] 1-3
                                                                                       *3   apfs_vnop_pagein + 300 (com.apple.filesystems.apfs + 458668) [0xfffffe000b57de1c] 1-3
                                                                                         *3   apfs_pagein + 2004 (com.apple.filesystems.apfs + 460936) [0xfffffe000b57e6f8] 1-3
                                                                                           *3   cluster_pagein + 28 (kernel.release.t8112 + 2238300) [0xfffffe000878275c] 1-3
                                                                                             *3   cluster_pagein_ext + 244 (kernel.release.t8112 + 2238556) [0xfffffe000878285c] 1-3
                                                                                               *3   ??? (kernel.release.t8112 + 2233840) [0xfffffe00087815f0] 1-3
                                                                                                 *3   buf_biowait + 104 (kernel.release.t8112 + 2188596) [0xfffffe0008776534] 1-3
                                                                                                   *3   ??? (kernel.release.t8112 + 5566384) [0xfffffe0008aaefb0] 1-3
                                                                                                     *3   lck_mtx_sleep + 228 (kernel.release.t8112 + 471164) [0xfffffe00085d307c] 1-3
                                                                                                       *3   ??? (kernel.release.t8112 + 561416) [0xfffffe00085e9108] 1-3
                                                                                                         *3   ??? (kernel.release.t8112 + 567968) [0xfffffe00085eaaa0] 1-3
                                                              1   swift_arrayDestroy + 124 (libswiftCore.dylib + 3659640) [0x191ffa778] 4
                                                                1   bool swift::RefCounts<swift::RefCountBitsT<(swift::RefCountInlinedness)1>>::doDecrementSlow<(swift::PerformDeinit)1>(swift::RefCountBitsT<(swift::RefCountInlinedness)1>, unsigned int) + 136 (libswiftCore.dylib + 3742348) [0x19200ea8c] 4
                                                                  1   _swift_release_dealloc + 56 (libswiftCore.dylib + 3739056) [0x19200ddb0] 4
                                                                    1   free_medium + 3244 (libsystem_malloc.dylib + 38880) [0x1821cb7e0] 4
                                                                      1   medium_free_scan_madvise_free + 424 (libsystem_malloc.dylib + 172128) [0x1821ec060] 4
                                                                        1   madvise + 8 (libsystem_kernel.dylib + 17656) [0x1823704f8] 4
                                                                         *1   ??? (kernel.release.t8112 + 30924) [0xfffffe00085678cc] 4
                                                                           *1   ??? (kernel.release.t8112 + 1678792) [0xfffffe00086f9dc8] 4
                                                                             *1   ??? (kernel.release.t8112 + 6712544) [0xfffffe0008bc6ce0] 4
                                                                               *1   ??? (kernel.release.t8112 + 5453268) [0xfffffe0008a935d4] 4
                                                                                 *1   ??? (kernel.release.t8112 + 1197420) [0xfffffe000868456c] 4
                                                                                   *1   ??? (kernel.release.t8112 + 1236324) [0xfffffe000868dd64] 4
                                                                                     *1   ??? (kernel.release.t8112 + 1010260) [0xfffffe0008656a54] 4
                                                                                       *1   ??? (kernel.release.t8112 + 1001496) [0xfffffe0008654818] 4
                                                                                         *1   ??? (kernel.release.t8112 + 1001108) [0xfffffe0008654694] 4
                                                                                           *1   ??? (kernel.release.t8112 + 1001108) [0xfffffe0008654694] (running) 4
                                                              1   swift_release + 48 (libswiftCore.dylib + 3727072) [0x19200aee0] 5
                                                               *1   ??? (kernel.release.t8112 + 30924) [0xfffffe00085678cc] 5
                                                                 *1   ??? (kernel.release.t8112 + 1679204) [0xfffffe00086f9f64] 5
                                                                   *1   ??? (kernel.release.t8112 + 1681884) [0xfffffe00086fa9dc] 5
                                                                     *1   ??? (kernel.release.t8112 + 1047444) [0xfffffe000865fb94] 5
                                                                       *1   ??? (kernel.release.t8112 + 1064764) [0xfffffe0008663f3c] 5
                                                                         *1   ??? (kernel.release.t8112 + 1009828) [0xfffffe00086568a4] 5
                                                                           *1   ??? (kernel.release.t8112 + 998824) [0xfffffe0008653da8] 5
                                                                             *1   ??? (kernel.release.t8112 + 999652) [0xfffffe00086540e4] 5
                                                                               *1   ??? (kernel.release.t8112 + 992140) [0xfffffe000865238c] 5
                                                                                 *1   ??? (kernel.release.t8112 + 1029292) [0xfffffe000865b4ac] 5
                                                                                   *1   ??? (kernel.release.t8112 + 6180976) [0xfffffe0008b45070] 5
                                                                                     *1   ??? (kernel.release.t8112 + 6153540) [0xfffffe0008b3e544] 5
                                                                                       *1   apfs_vnop_pagein + 300 (com.apple.filesystems.apfs + 458668) [0xfffffe000b57de1c] 5
                                                                                         *1   apfs_pagein + 2004 (com.apple.filesystems.apfs + 460936) [0xfffffe000b57e6f8] 5
                                                                                           *1   cluster_pagein + 28 (kernel.release.t8112 + 2238300) [0xfffffe000878275c] 5
                                                                                             *1   cluster_pagein_ext + 244 (kernel.release.t8112 + 2238556) [0xfffffe000878285c] 5
                                                                                               *1   ??? (kernel.release.t8112 + 2233840) [0xfffffe00087815f0] 5
                                                                                                 *1   buf_biowait + 104 (kernel.release.t8112 + 2188596) [0xfffffe0008776534] 5
                                                                                                   *1   ??? (kernel.release.t8112 + 5566384) [0xfffffe0008aaefb0] 5
                                                                                                     *1   lck_mtx_sleep + 228 (kernel.release.t8112 + 471164) [0xfffffe00085d307c] 5
                                                                                                       *1   ??? (kernel.release.t8112 + 561416) [0xfffffe00085e9108] 5
                                                                                                         *1   ??? (kernel.release.t8112 + 567968) [0xfffffe00085eaaa0] 5
                                                              2   swift_arrayDestroy + 124 (libswiftCore.dylib + 3659640) [0x191ffa778] 6-7
                                                                2   bool swift::RefCounts<swift::RefCountBitsT<(swift::RefCountInlinedness)1>>::doDecrementSlow<(swift::PerformDeinit)1>(swift::RefCountBitsT<(swift::RefCountInlinedness)1>, unsigned int) + 136 (libswiftCore.dylib + 3742348) [0x19200ea8c] 6-7
                                                                  2   _swift_release_dealloc + 56 (libswiftCore.dylib + 3739056) [0x19200ddb0] 6-7
                                                                    2   free_medium + 3244 (libsystem_malloc.dylib + 38880) [0x1821cb7e0] 6-7
                                                                      1   medium_free_scan_madvise_free + 424 (libsystem_malloc.dylib + 172128) [0x1821ec060] 6
                                                                        1   madvise + 8 (libsystem_kernel.dylib + 17656) [0x1823704f8] 6
                                                                         *1   ??? (kernel.release.t8112 + 30924) [0xfffffe00085678cc] 6
                                                                           *1   ??? (kernel.release.t8112 + 1678792) [0xfffffe00086f9dc8] 6
                                                                             *1   ??? (kernel.release.t8112 + 6712544) [0xfffffe0008bc6ce0] 6
                                                                               *1   ??? (kernel.release.t8112 + 5453268) [0xfffffe0008a935d4] 6
                                                                                 *1   ??? (kernel.release.t8112 + 1197420) [0xfffffe000868456c] 6
                                                                                   *1   ??? (kernel.release.t8112 + 1236448) [0xfffffe000868dde0] 6
                                                                                     *1   ??? (kernel.release.t8112 + 1618696) [0xfffffe00086eb308] 6
                                                                                       *1   ??? (kernel.release.t8112 + 1618696) [0xfffffe00086eb308] (running) 6
                                                                      1   medium_free_scan_madvise_free + 204 (libsystem_malloc.dylib + 171908) [0x1821ebf84] 7
                                                                       *1   ??? (kernel.release.t8112 + 30924) [0xfffffe00085678cc] 7
                                                                         *1   ??? (kernel.release.t8112 + 1679204) [0xfffffe00086f9f64] 7
                                                                           *1   ??? (kernel.release.t8112 + 1681884) [0xfffffe00086fa9dc] 7
                                                                             *1   ??? (kernel.release.t8112 + 1047444) [0xfffffe000865fb94] 7
                                                                               *1   ??? (kernel.release.t8112 + 1064764) [0xfffffe0008663f3c] 7
                                                                                 *1   ??? (kernel.release.t8112 + 1009828) [0xfffffe00086568a4] 7
                                                                                   *1   ??? (kernel.release.t8112 + 998824) [0xfffffe0008653da8] 7
                                                                                     *1   ??? (kernel.release.t8112 + 999652) [0xfffffe00086540e4] 7
                                                                                       *1   ??? (kernel.release.t8112 + 992140) [0xfffffe000865238c] 7
                                                                                         *1   ??? (kernel.release.t8112 + 1029292) [0xfffffe000865b4ac] 7
                                                                                           *1   ??? (kernel.release.t8112 + 6180976) [0xfffffe0008b45070] 7
                                                                                             *1   ??? (kernel.release.t8112 + 6153540) [0xfffffe0008b3e544] 7
                                                                                               *1   apfs_vnop_pagein + 300 (com.apple.filesystems.apfs + 458668) [0xfffffe000b57de1c] 7
                                                                                                 *1   apfs_pagein + 2004 (com.apple.filesystems.apfs + 460936) [0xfffffe000b57e6f8] 7
                                                                                                   *1   cluster_pagein + 28 (kernel.release.t8112 + 2238300) [0xfffffe000878275c] 7
                                                                                                     *1   cluster_pagein_ext + 244 (kernel.release.t8112 + 2238556) [0xfffffe000878285c] 7
                                                                                                       *1   ??? (kernel.release.t8112 + 2233840) [0xfffffe00087815f0] 7
                                                                                                         *1   buf_biowait + 104 (kernel.release.t8112 + 2188596) [0xfffffe0008776534] 7
                                                                                                           *1   ??? (kernel.release.t8112 + 5566384) [0xfffffe0008aaefb0] 7
                                                                                                             *1   lck_mtx_sleep + 228 (kernel.release.t8112 + 471164) [0xfffffe00085d307c] 7
                                                                                                               *1   ??? (kernel.release.t8112 + 561416) [0xfffffe00085e9108] 7
                                                                                                                 *1   ??? (kernel.release.t8112 + 567968) [0xfffffe00085eaaa0] 7
                                                              4   swift_release + 48 (libswiftCore.dylib + 3727072) [0x19200aee0] 8-11
                                                               *4   ??? (kernel.release.t8112 + 30924) [0xfffffe00085678cc] 8-11
                                                                 *4   ??? (kernel.release.t8112 + 1679204) [0xfffffe00086f9f64] 8-11
                                                                   *4   ??? (kernel.release.t8112 + 1681884) [0xfffffe00086fa9dc] 8-11
                                                                     *4   ??? (kernel.release.t8112 + 1047444) [0xfffffe000865fb94] 8-11
                                                                       *4   ??? (kernel.release.t8112 + 1064764) [0xfffffe0008663f3c] 8-11
                                                                         *4   ??? (kernel.release.t8112 + 1009828) [0xfffffe00086568a4] 8-11
                                                                           *4   ??? (kernel.release.t8112 + 998824) [0xfffffe0008653da8] 8-11
                                                                             *4   ??? (kernel.release.t8112 + 999652) [0xfffffe00086540e4] 8-11
                                                                               *4   ??? (kernel.release.t8112 + 992140) [0xfffffe000865238c] 8-11
                                                                                 *4   ??? (kernel.release.t8112 + 1029292) [0xfffffe000865b4ac] 8-11
                                                                                   *4   ??? (kernel.release.t8112 + 6180976) [0xfffffe0008b45070] 8-11
                                                                                     *4   ??? (kernel.release.t8112 + 6153540) [0xfffffe0008b3e544] 8-11
                                                                                       *4   apfs_vnop_pagein + 300 (com.apple.filesystems.apfs + 458668) [0xfffffe000b57de1c] 8-11
                                                                                         *4   apfs_pagein + 2004 (com.apple.filesystems.apfs + 460936) [0xfffffe000b57e6f8] 8-11
                                                                                           *4   cluster_pagein + 28 (kernel.release.t8112 + 2238300) [0xfffffe000878275c] 8-11
                                                                                             *4   cluster_pagein_ext + 244 (kernel.release.t8112 + 2238556) [0xfffffe000878285c] 8-11
                                                                                               *4   ??? (kernel.release.t8112 + 2233840) [0xfffffe00087815f0] 8-11
                                                                                                 *4   buf_biowait + 104 (kernel.release.t8112 + 2188596) [0xfffffe0008776534] 8-11
                                                                                                   *4   ??? (kernel.release.t8112 + 5566384) [0xfffffe0008aaefb0] 8-11
                                                                                                     *4   lck_mtx_sleep + 228 (kernel.release.t8112 + 471164) [0xfffffe00085d307c] 8-11
                                                                                                       *4   ??? (kernel.release.t8112 + 561416) [0xfffffe00085e9108] 8-11
                                                                                                         *4   ??? (kernel.release.t8112 + 567968) [0xfffffe00085eaaa0] 8-11

  Thread 0x1fbf    Swift Task UNKNOWN    DispatchQueue "com.apple.main-thread"(1)    11 samples (1-11)    priority 46 (base 46)    cpu time 0.108s (330.3M cycles, 607.7M instructions, 0.54c/i)
  <thread QoS user interactive (requested user interactive), process unclamped, process received importance donation from WindowServer [567], process received importance donation from distnoted [876], process received importance donation from filecoordinationd [1028], IO tier 0>
  11  <truncated backtrace> 1-11
    11  ??? (CotEditor + 423429) [0x100933605] 1-11
      11  ??? (CotEditor + 2609677) [0x100b4920d] 1-11
        11  ??? (CotEditor + 423353) [0x1009335b9] 1-11
          3   <callstack in another call tree for Thread 0x1fbf: ??? (CotEditor + 381440) [0x100929200]> 1-3
          1   <callstack in another call tree for Thread 0x1fbf: ??? (CotEditor + 381440) [0x100929200]> (running) 4
          1   <callstack in another call tree for Thread 0x1fbf: ??? (CotEditor + 381440) [0x100929200]> 5
          1   <callstack in another call tree for Thread 0x1fbf: ??? (CotEditor + 381440) [0x100929200]> (running) 6
          5   <callstack in another call tree for Thread 0x1fbf: ??? (CotEditor + 381440) [0x100929200]> 7-11

  Thread 0x254e    Thread name "com.apple.NSEventThread"    11 samples (1-11)    priority 46 (base 46)
  <thread QoS user interactive (requested user interactive), process unclamped, process received importance donation from WindowServer [567], process received importance donation from distnoted [876], process received importance donation from filecoordinationd [1028], IO tier 0>
  11  thread_start + 8 (libsystem_pthread.dylib + 7740) [0x1823a8e3c] 1-11
    11  _pthread_start + 136 (libsystem_pthread.dylib + 28724) [0x1823ae034] 1-11
      11  _NSEventThread + 144 (AppKit + 1454432) [0x185d92160] 1-11
        11  CFRunLoopRunSpecific + 608 (CoreFoundation + 506172) [0x18248993c] 1-11
          11  __CFRunLoopRun + 1208 (CoreFoundation + 509004) [0x18248a44c] 1-11
            11  __CFRunLoopServiceMachPort + 160 (CoreFoundation + 514948) [0x18248bb84] 1-11
              11  mach_msg + 24 (libsystem_kernel.dylib + 7160) [0x18236dbf8] 1-11
                11  mach_msg_overwrite + 476 (libsystem_kernel.dylib + 42160) [0x1823764b0] 1-11
                  11  mach_msg2_trap + 8 (libsystem_kernel.dylib + 6260) [0x18236d874] 1-11
                   *11  ??? (kernel.release.t8112 + 226340) [0xfffffe0008597424] 1-11

  Binary Images:
           0x1008cc000 -        0x100c9bfff  com.coteditor.CotEditor 4.7.3 (627)        <AE6A14B2-CEB3-3232-8218-26C78637E73A>             /Applications/CotEditor.app/Contents/MacOS/CotEditor
           0x182027000 -        0x1820bb387  dyld (1125.5)                              <50746901-DB0E-39A0-B391-BAAA6B82AD0F>             /usr/lib/dyld
           0x1821c2000 -        0x1821f8fff  libsystem_malloc.dylib (474.0.13)          <FAA92B9B-9273-3C30-BB9E-36346ECA2FAF>             /usr/lib/system/libsystem_malloc.dylib
           0x1821f9000 -        0x18223ffff  libdispatch.dylib (1462.80.1)              <5AA1649C-EF1D-39F7-A66C-4C5D2E53C474>             /usr/lib/system/libdispatch.dylib
           0x18236c000 -        0x1823a6fff  libsystem_kernel.dylib (10002.81.5)        <A7228B5D-53C7-3FE9-84E4-2A8C04DCF051>             /usr/lib/system/libsystem_kernel.dylib
           0x1823a7000 -        0x1823b3ff3  libsystem_pthread.dylib (519)              <449BBAD3-F7EF-371D-9A59-FD4FFA78289B>             /usr/lib/system/libsystem_pthread.dylib
           0x18240e000 -        0x1828e5fff  com.apple.CoreFoundation 6.9 (2303)        <6030A572-B731-3F46-B3A0-2598FBB98C9A>             /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
           0x185c2f000 -        0x186f5afff  com.apple.AppKit 6.9 (2487.40.107)         <873E5DA0-BC01-399B-BF60-4066922D6A17>             /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
           0x18ca22000 -        0x18cce1fff  com.apple.HIToolbox 2.1.1 (1202)           <DB8A8B63-7BFB-384B-B0A9-C124675FBB39>             /System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/HIToolbox
           0x191c7d000 -        0x1921bdfff  libswiftCore.dylib (5.9.2.2.12)            <7EBA7EF0-57D2-3155-B45D-1A4D40A732F3>             /usr/lib/swift/libswiftCore.dylib
           0x2235c8000 -        0x223626ff3  libswift_Concurrency.dylib (5.9.2.2.12)    <2EB0DF2B-7B1D-3D6D-90F1-9F8CC621B18F>             /usr/lib/swift/libswift_Concurrency.dylib
   *0xfffffe0008560000 - 0xfffffe0008db7fff  kernel.release.t8112 (10002.81.5)          <C7428EB0-BE7C-3056-8267-A75FED2CC4A4>__TEXT_EXEC  /System/Library/Kernels/kernel.release.t8112
   *0xfffffe000b50de70 - 0xfffffe000b66f59b  com.apple.filesystems.apfs (2235.80.4.0.1) <B003ABBF-5D1E-3751-8BDB-A6FC6289E94E>__TEXT_EXEC  com.apple.filesystems.APFS
```
